blocks/identity-block/components/SocialSignOn/index.jsx

```diff
@@ -0,0 +1,107 @@
+import React, { useEffect, useState, useRef } from 'react';
+import { isServerSide } from '@wpmedia/engine-theme-sdk';
+import PropTypes from '@arc-fusion/prop-types';
+import useIdentity from '../Identity';
+import './styles.scss';
+
+const SocialSignOn = ({ onError, redirectURL }) => {
+  const { Identity } = useIdentity();
+  const [config, setConfig] = useState(() => Identity?.configOptions ?? {});
+  const [isGoogleInitialized, setIsGoogleInitialized] = useState(false);
+  const [isFacebookInitialized, setIsFacebookInitialized] = useState(false);
+
+  const isGoogleReset = useRef(false);
+
+  if (!isServerSide() && !window.onFacebookSignOn) {
+    window.onFacebookSignOn = async () => {
+      try {
+        await Identity.facebookSignOn();
+        window.location = redirectURL;
+      } catch (e) {
+        onError();
+      }
+    };
+  }
+
+  useEffect(() => {
+    const fetchConfig = async () => {
+      await Identity.getConfig();
+      setConfig(Identity.configOptions);
+    };
+    if (Identity && !Identity.configOptions) {
+      fetchConfig();
+    }
+  }, [Identity]);
+
+  useEffect(() => {
+    const initializeFacebook = async () => {
+      await Identity.initFacebookLogin(null);
+    };
+    const initializeGoogle = async () => {
+      await Identity.initGoogleLogin(null, {
+        width: 300,
+        height: 48,
+        onSuccess: () => {
+          if (!isGoogleReset.current) {
+            isGoogleReset.current = true;
+          } else {
+            window.location = redirectURL;
+          }
+        },
+        onFailure: () => { onError(); },
+      });
+    };
+    if (config.googleClientId && !isGoogleInitialized) {
+      initializeGoogle();
+      setIsGoogleInitialized(true);
+    }
+    if (config.facebookAppId && !isFacebookInitialized) {
+      initializeFacebook();
+      setIsFacebookInitialized(true);
+    }
+  }, [Identity,
+    config.facebookAppId,
+    config.googleClientId,
+    isFacebookInitialized,
+    isGoogleInitialized,
+    onError,
+    redirectURL]);
+
+  if (!Identity) {
+    return null;
+  }
+
+  if (!config.facebookAppId && !config.googleClientId) {
+    return null;
+  }
+
+  return (
+    <section className="xpmedia-social-signin-wrapper">
+      {
+        config.googleClientId ? <div id="google-sign-in-button" /> : null
+      }
+      {
+        config.facebookAppId ? (
```

Comment:
if no id found, throw error? seems to fail silently, we discovered if no google id button. is there ac for this?

Reply:
I think this is fine?  If there is no facebook app id defined there's no way we can render the facebook login because the login needs that id.